meta-mender-tegra/scripts/mender_tegra_test.py

```diff
@@ -0,0 +1,327 @@
+            raise RuntimeError("Missing argument install")
+        result = conn.run("mender -install {}".format(args.install))
+        # Mender doesn't return error states, so we can't check return codes here
+        match = re.search(r' level=error ', result.stderr, re.MULTILINE)
+        if match is not None:
+            raise RuntimeError("Mender install failed with error messages in the logs")
+
+    def add_sentinel_file(self):
+        """
+        Generate a sentinel file signifying not to mark the boot as successful to test rollback cases.
+        The meta-mender-tegra layer has a service which does not mark the boot as successful if the file exists.
+        This is used to test rollback scenarios.
+        """
+        conn = self.get_connection()
+        conn.run("mkdir -p /var/lib/mender")
+        conn.run("touch /var/lib/mender/dont-mark-next-boot-successful")
+
+    def remove_sentinel_file(self):
+        '''
+        Remove the sentinel file for normal operation i.e successful boot.
+        '''
+        conn = self.get_connection()
+        conn.run("rm -f /var/lib/mender/dont-mark-next-boot-successful")
+
+    def mender_commit(self):
+        '''
+        Commit changes for mender updates after a reboot.
+        '''
+        self.get_connection().run("mender -commit")
+
+    def reboot(self):
+        '''
+        Function reboot the device and establish ssh connection
+        '''
+        conn = self.get_connection()
+        print("Rebooting device")
+        result = conn.run("reboot", warn=True)
+        self.wait_for_device_removal()
+        self.wait_for_device()
+
+    def nvbootctrl_current_slot(self):
+        '''
+        Returns the current boot slot value
+        '''
+        conn = self.get_connection()
+        result = conn.run("nvbootctrl get-current-slot")
+        return result.stdout.strip()
+
+    def check_partition_mismatch(self):
+        '''
+        Function to check boot and rootfs partition mismatch after reboot.
+        To be used after every reboot.
+        '''
+        conn = self.get_connection()
+        boot_slot = self.nvbootctrl_current_slot()
+        rootfs_part = None
+        if boot_slot == '0':
+            rootfs_part = 'A'
+        elif boot_slot == '1':
+            rootfs_part = 'B'
+
+        if rootfs_part != None:
+            result = conn.run(f'grep -h {rootfs_part} /etc/mender/mender.conf /var/lib/mender/mender.conf | cut -d: -f2 | cut -d, -f1 | tr -d \'" \'')
+            result = conn.run(f'df -h | grep {result.stdout.strip()}')
+            if result.return_code != 0:
+                raise RuntimeError("Boot and Rootfs Partition Mismatch detected")
+        else:
+            raise RuntimeError("Cannot Identify Rootfs Partition Slot")
+
+    def reboot_and_verify_slot(self):
+        '''
+        Verifies if the boot slot has not changed after a reboot.
+        Function to be used when we want to verify if simple reboots lead to boot slot change.
```

Comment:
Function to be used when we want to verify simple reboots do not lead to boot slot change?